hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -197,22 +198,25 @@ async def _get_account_id(self) -> int:
         params = {
             "OMSId": 1,
             "UserId": int(self._auth.uid),
-            "UserName": self._auth.username
+            "UserName": self._auth.account_name
         }
 
         resp: List[int] = await self._api_request(
             "GET",
-            path_url=CONSTANTS.USER_ACCOUNTS_PATH_URL,
+            path_url=CONSTANTS.USER_ACCOUNT_INFOS_PATH_URL,
             params=params,
             is_auth_required=True,
         )
 
-        """
-        NOTE: Currently there is no way to determine which accountId the user intends to use.
-              The GetUserAccountInfos endpoint doesnt seem to provide anything useful either.
-              The assumption here is that the FIRST entry in the list is the accountId the user intends to use
-        """
-        return resp[0]
+        account_info = next((account_info for account_info in resp
+                             if account_info.get("AccountName") == self._auth.account_name),
+                            None)
+        if account_info is None:
+            self.logger().error(f"There is no account named {self._auth.account_name} "
+                                f"associated with the current NDAX user")
+            raise ValueError(f"NDAX account named {self._auth.account_name} does not exist")
```

Comment:
Not sure if this `ValueError` will be raised. I figure we dont need to raise an error here. Since in `ready`, there are checks to ensure that the `account_id` is initialized. If it is not initialized, it will not start, which is an expected behaviour.